lib/internal/modules/run_main.js

```diff
@@ -166,9 +171,7 @@ function executeUserEntryPoint(main = process.argv[1]) {
     wrapModuleLoad(main, null, true);
   } else {
     const mainPath = resolvedMain || main;
-    if (mainURL === undefined) {
-      mainURL = pathToFileURL(mainPath).href;
-    }
+    const mainURL = getOptionValue('--entry-url') ? new URL(mainPath, getCWDURL()) : pathToFileURL(mainPath);
```

Comment:
I agree with omitting `.href` here if it works, and it doesn't seem to cause any test failures, but i'm not sure if the typings there are aligned with this.
If i'm tracing the following `cascadedLoader.import(mainURL, ...)` correctly, its first parameter is defined as `string`: https://github.com/nodejs/node/blob/779e6bdd5ebad6c0558401e6ee5da8a5af476f47/lib/internal/modules/esm/loader.js#L532-L540

Reply:
Omitted `href` works, and IMHO changing that is an issue for a different PR, because, while correct, it's unrelated to the actual `--entry-url` flag.

Reply:
A specifier can be anything that can be coerced into a string. I'm not sure the typings are really incorrect, but in any case, passing a `URL` definitely works.